### ISSUE TITLE
Can't login with `cosmic-greeter`

### ISSUE THREAD

**Issue by futile** [lilyinstarlight/nixos-cosmic#285]
Hi,

Using NixOS unstable on x86_64.

when I enable cosmic-greeter and reboot, it starts, but with some weirdness:
* The user selection contains one entry, with a checkmark in front of it, but there is no username displayed. So it's an empty label with a checkmark next to it.
* The keyboard layout drop-down is empty.

When I then input my password and try to login, I get an error that the login failed (even though it was the correct password).

Here is my journalctl log from rebooting and trying to log in (and having to reboot again):

```console
$ journalctl --boot=-1 | rg 'cosmic-greeter|greetd'
Aug 28 16:41:56 nixos-home systemd[1]: Starting cosmic-greeter-daemon.service...
Aug 28 16:41:56 nixos-home systemd[1]: Started cosmic-greeter-daemon.service.
Aug 28 16:41:57 nixos-home systemd[1]: Started greetd.service.
Aug 28 16:42:04 nixos-home greetd[2156]: pam_unix(greetd:session): session opened for user cosmic-greeter(uid=991) by (uid=0)
Aug 28 16:42:04 nixos-home greetd[2156]: pam_systemd(greetd:session): New sd-bus connection (system-bus-pam-systemd-2156) opened.
Aug 28 16:42:04 nixos-home systemd-logind[1339]: New session 1 of user cosmic-greeter.
Aug 28 16:42:04 nixos-home (systemd)[2170]: pam_unix(systemd-user:session): session opened for user cosmic-greeter(uid=991) by (uid=0)
Aug 28 16:42:04 nixos-home systemd-logind[1339]: New session 2 of user cosmic-greeter.
Aug 28 16:42:04 nixos-home systemd[1]: Started Session 1 of User cosmic-greeter.
Aug 28 16:42:04 nixos-home greetd[2156]: gkr-pam: gnome-keyring-daemon started properly
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.602320Z ERROR cosmic_comp::config: Failed to read config 'workspaces' err=NoConfigDirectory
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.608185Z ERROR cosmic_settings_config::shortcuts: failed to read local shortcuts config 'system_actions': NoConfigDirectory
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.609970Z  WARN cosmic_settings_config::shortcuts::sym: Key-Binding 'Period' only matched case insensitive for "period"
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.609998Z  WARN cosmic_settings_config::shortcuts::sym: Key-Binding 'Comma' only matched case insensitive for "comma"
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.610006Z  WARN cosmic_settings_config::shortcuts::sym: Key-Binding 'Period' only matched case insensitive for "period"
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.610012Z  WARN cosmic_settings_config::shortcuts::sym: Key-Binding 'Comma' only matched case insensitive for "comma"
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.610072Z ERROR cosmic_settings_config::shortcuts: shortcuts custom config error: GetKey("custom", Os { code: 2, kind: NotFound, message: "No such file or directory" })
Aug 28 16:42:04 nixos-home cosmic-greeter[2207]: 2024-08-28T14:42:04.791978Z  WARN smithay::backend::drm::device::fd: Unable to become drm master, assuming unprivileged mode
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vwv398fqpy177ahwazfdp7z9wls5d6xf-gnome-mimeapps/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/var/lib/cosmic-greeter/.nix-profile/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/profile/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/var/lib/cosmic-greeter/.local/state/nix/profile/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/etc/profiles/per-user/cosmic-greeter/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/var/nix/profiles/default/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] session COSMIC overwrote old command (["/nix/store/w1iq3315z63558j04gnlzdd2yk1v1hfz-coreutils-9.5/bin/env", "XDG_SESSION_TYPE=wayland", "XDG_CURRENT_DESKTOP=COSMIC", "XDG_SESSION_DESKTOP=COSMIC", "/nix/store/xwm917rh8j7rd397glpgypmf78qls80j-cosmic-session-1.0.0-alpha.1-unstable-2024-07-31/bin/start-cosmic"], ["XDG_SESSION_TYPE=wayland", "XDG_CURRENT_DESKTOP=COSMIC", "XDG_SESSION_DESKTOP=COSMIC"])
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/wl1bnmf2766zyq23j2d0lp0xm0f92p38-gnome-shell-46.4/share/gsettings-schemas/gnome-shell-46.4/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/5lfwapng73i9km4b3n95mp2ldjlvvj65-gnome-shell-extensions-46.2/share/gsettings-schemas/gnome-shell-extensions-46.2/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/11fhnhcfa1k8fq2kq8hv7srs8iqiz7bq-cosmic-comp-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vy9flz7dfrd7q47agx4zg2p8i1h1c52x-cosmic-settings-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/4m60l1l1aa42nxwqbbc43lqpgzw14xhs-cosmic-icons-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/6qlrlcx40rgpgrmf687jlgxi9jx75xrb-cosmic-greeter-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vy9flz7dfrd7q47agx4zg2p8i1h1c52x-cosmic-settings-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/4m60l1l1aa42nxwqbbc43lqpgzw14xhs-cosmic-icons-1.0.0-alpha.1-unstable-2024-08-15/share/wayland-sessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vwv398fqpy177ahwazfdp7z9wls5d6xf-gnome-mimeapps/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] session GNOME overwrote old command (["/nix/store/w1iq3315z63558j04gnlzdd2yk1v1hfz-coreutils-9.5/bin/env", "XDG_SESSION_TYPE=wayland", "XDG_CURRENT_DESKTOP=GNOME", "XDG_SESSION_DESKTOP=GNOME", "/nix/store/0cdmh04qpbpz0jgr8q8i5m2l02hdxrw6-gnome-session-46.0/bin/gnome-session"], ["XDG_SESSION_TYPE=wayland", "XDG_CURRENT_DESKTOP=GNOME", "XDG_SESSION_DESKTOP=GNOME"])
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/var/lib/cosmic-greeter/.nix-profile/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/profile/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/var/lib/cosmic-greeter/.local/state/nix/profile/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/etc/profiles/per-user/cosmic-greeter/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/var/nix/profiles/default/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/run/current-system/sw/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/wl1bnmf2766zyq23j2d0lp0xm0f92p38-gnome-shell-46.4/share/gsettings-schemas/gnome-shell-46.4/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/5lfwapng73i9km4b3n95mp2ldjlvvj65-gnome-shell-extensions-46.2/share/gsettings-schemas/gnome-shell-extensions-46.2/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/11fhnhcfa1k8fq2kq8hv7srs8iqiz7bq-cosmic-comp-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vy9flz7dfrd7q47agx4zg2p8i1h1c52x-cosmic-settings-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/4m60l1l1aa42nxwqbbc43lqpgzw14xhs-cosmic-icons-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/6qlrlcx40rgpgrmf687jlgxi9jx75xrb-cosmic-greeter-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/vy9flz7dfrd7q47agx4zg2p8i1h1c52x-cosmic-settings-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to read session directory "/nix/store/4m60l1l1aa42nxwqbbc43lqpgzw14xhs-cosmic-icons-1.0.0-alpha.1-unstable-2024-08-15/share/xsessions": Os { code: 2, kind: NotFound, message: "No such file or directory" }
Aug 28 16:42:05 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:05Z WARN  cosmic_greeter::greeter] failed to load keyboard layouts: No such file or directory (os error 2)
Aug 28 16:42:12 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:12Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:12 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:12Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:13 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:13Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:13 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:13Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:21 nixos-home greetd[2251]: pam_unix(greetd:auth): authentication failure; logname= uid=0 euid=0 tty= ruser= rhost=  user=felix
Aug 28 16:42:21 nixos-home greetd[2251]: gkr-pam: unable to locate daemon control file
Aug 28 16:42:21 nixos-home greetd[2251]: gkr-pam: stashed password to try later in open session
Aug 28 16:42:23 nixos-home greetd[2251]: error: authentication error: pam_authenticate: AUTH_ERR
Aug 28 16:42:23 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:23Z WARN  cosmic_greeter::greeter::ipc] error while cancelling session: unable to send message: Connection refused (os error 111)
Aug 28 16:42:28 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:28Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:28 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:28Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:28 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:28Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:35 nixos-home greetd[2657]: pam_unix(greetd:auth): authentication failure; logname= uid=0 euid=0 tty= ruser= rhost=  user=felix
Aug 28 16:42:35 nixos-home greetd[2657]: gkr-pam: unable to locate daemon control file
Aug 28 16:42:35 nixos-home greetd[2657]: gkr-pam: stashed password to try later in open session
Aug 28 16:42:37 nixos-home greetd[2657]: error: authentication error: pam_authenticate: AUTH_ERR
Aug 28 16:42:37 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:37Z WARN  cosmic_greeter::greeter::ipc] error while cancelling session: unable to send message: Connection refused (os error 111)
Aug 28 16:42:40 nixos-home cosmic-greeter[2736]: could not read cache file
Aug 28 16:42:40 nixos-home cosmic-greeter[2736]: No such file or directory (os error 2)
Aug 28 16:42:41 nixos-home cosmic-greeter[2838]: could not read cache file
Aug 28 16:42:41 nixos-home cosmic-greeter[2838]: No such file or directory (os error 2)
Aug 28 16:42:41 nixos-home cosmic-greeter[2862]: could not read cache file
Aug 28 16:42:41 nixos-home cosmic-greeter[2862]: No such file or directory (os error 2)
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:48 nixos-home cosmic-greeter[2238]: [2024-08-28T14:42:48Z WARN  tiny_skia::painter] empty paths and horizontal/vertical lines cannot be filled
Aug 28 16:42:52 nixos-home systemd[1]: Stopping Session 1 of User cosmic-greeter...
Aug 28 16:42:52 nixos-home greetd[1836]: error: check_children: greeter exited without creating a session
Aug 28 16:42:52 nixos-home systemd[1]: Stopping greetd.service...
Aug 28 16:42:52 nixos-home systemd[1]: greetd.service: Deactivated successfully.
Aug 28 16:42:52 nixos-home systemd[1]: Stopped greetd.service.
Aug 28 16:42:52 nixos-home systemd[1]: Stopped Session 1 of User cosmic-greeter.
Aug 28 16:42:52 nixos-home systemd[1]: Stopping cosmic-greeter-daemon.service...
Aug 28 16:42:52 nixos-home systemd[1]: cosmic-greeter-daemon.service: Deactivated successfully.
Aug 28 16:42:52 nixos-home systemd[1]: Stopped cosmic-greeter-daemon.service.
Aug 28 16:42:52 nixos-home (sd-pam)[2172]: pam_unix(systemd-user:session): session closed for user cosmic-greeter
```

Thanks a lot for taking a look!

**Comment by lilyinstarlight** [comment]
You wouldn't happen to have a keyboard layout set would you?

Username is not empty when it goes to PAM according to your log, but it did have a standard PAM authentication failure. But keyboard layouts are still busted due to hardcoded paths until the upstream fix makes it into the package

**Comment by lilyinstarlight** [comment]
Are you still having issues after updating?

If so, please provide output from running `getent passwd felix` in terminal

**Comment by futile** [comment]
Hey! So, I'm on a 2-3 days old commit of this repo, and it's working better now.

I still don't see my (non-QWERTY) keyboard layout or username, but when I type my password using QWERTY instead I can log in! Clicking the eye-symbol to reveal my password doesn't work, but login does.

Output of `getent` anyway:

```console
❯ getent passwd felix
felix:x:1000:100::/home/felix:/run/current-system/sw/bin/fish
```

**Comment by lilyinstarlight** [comment]
Ah well the output of your `getent` indicates your user's name is empty, so that solves that mystery. Upstream should probably show username when GECOS/name field is empty like that

Keyboard layout is still waiting on cosmic-greeter to bump xkb-data

**Comment by futile** [comment]
> Ah well the output of your `getent` indicates your user's name is empty, so that solves that mystery. Upstream should probably show username when GECOS/name field is empty like that

Ahh that makes sense, didn't even know there is a proper "Name" associated with usernames :D Thanks, and agree that the raw username should probably be a fallback!

> Keyboard layout is still waiting on cosmic-greeter to bump xkb-data

Also makes sense, thanks a lot! :)

We can close this and open a new issue for the "Name" fallback, since the login itself now works, or leave it open, whatever you prefer :)

Already thank you very much for taking a look and helping out! :)

**Comment by lilyinstarlight** [comment]
> We can close this and open a new issue for the "Name" fallback, since the login itself now works, or leave it open, whatever you prefer :)

It's an upstream issue, but I'll probably submit a fix for it myself so this can probably stay open

**Comment by lilyinstarlight** [comment]
The username issue is resolved, can you confirm if this is fine to be closed now? The xkb-data issue is being tracked by #74

**Comment by futile** [comment]
Yep, I can confirm that I can see my username now!

Apart from the xkb-data thing, clicking the eye-icon in the password field still doesn't do anything (neither does hovering it). Also, when cosmic-greeter starts, my mouse cursor comes with a weird graphics-glitch box next to it (like a tooltip, around 1/10th of my screen width), which disappears when I move the cursor over a ui element (like a button, input field, etc.).

Should I report these in a new ticket? Otherwise this is fine to be closed now, I'll let you have the honor :)

Thanks again for taking a look, contributing a fix, and staying in touch! :)

**Comment by lilyinstarlight** [comment]
> Should I report these in a new ticket? Otherwise this is fine to be closed now, I'll let you have the honor :)

Yeah the rest of these issues should probably be best reported upstream. Thanks for confirming!

If I have time, I might fix the password reveal thingy but that can be tracked upstream :)
